src/pet.js

```diff
@@ -0,0 +1,110 @@
+class Pet {    
+    constructor(name) {
+        this.name = name || "An unloved pet";
+        this.age = 0;
+        this.hunger = 0;
+        this.fitness = 10;
+        this.poop = 0;
+        this.children = [];
+    }
+
+    get isAlive() {
+        const MAX_AGE = 30;
+        const MAX_HUNGER = 10;
+        const MIN_FITNESS = 0;
+
+        return this.age < MAX_AGE && this.hunger < MAX_HUNGER && this.fitness > MIN_FITNESS;
+    }
+
+    get deadMessage() {
+        const DEAD_MESSAGE = 'Your pet is no longer alive :(';
+
+        return DEAD_MESSAGE;
+    }
+
+    growUp() {
+        const AGING_FACTOR = 1;
+        const HUNGER_FACTOR = 5;
+        const FITNESS_FACTOR = 3;
+        
+        if(!this.isAlive){
+            throw new Error(this.deadMessage);
+        }
+
+        this.age = this.age + AGING_FACTOR;
+        this.hunger = this.hunger + HUNGER_FACTOR;
+        this.fitness = this.fitness - FITNESS_FACTOR;
+    }
+
+    walk() {
+        const WALK_FACTOR = 4;
+        const MAX_FITNESS = 10;
+
+        if(!this.isAlive){
+            throw new Error(this.deadMessage);
+        }
+
+        this.fitness = this.fitness + WALK_FACTOR;
+
+        if (this.fitness > MAX_FITNESS) {
+            this.fitness = MAX_FITNESS;
+        } 
+    }
+
+    feed() {
+        const FEED_FACTOR = 3;
+        const POOP_FACTOR = 0.25;
+        const MIN_HUNGER = 0;
+
+        if(!this.isAlive){
```

Comment:
You use these lines quite a few times in the code, worth making it's own function?